bundler/lib/bundler/cli/exec.rb

```diff
@@ -18,8 +18,10 @@ def initialize(options, args)
     def run
       validate_cmd!
       SharedHelpers.set_bundle_environment
-      if bin_path = Bundler.which(cmd)
+      if (bin_path = Bundler.which(cmd))
         if !Bundler.settings[:disable_exec_load] && ruby_shebang?(bin_path)
+          # remove unnecessary relative path for `load`
+          bin_path.delete_prefix!("./")
```

Comment:
Are the changes to this file necessary?

Reply:
This is for passing this test
https://github.com/rubygems/rubygems/actions/runs/12035052468/job/33553094293

Reply:
Aaah I see! That makes me wonder, should instead leave `Bundler.which` untouched and instead prepend `./` to `bin_path` when `kernel_exec`'ing below in line 25?

Reply:
sure let's do that

Reply:
Cool, so `Bundle.which` implementation should be left untouched, but let's leave the new tests, they are much better.